### PR TITLE
feat: support mogdb physical backup & restore

### PR DESCRIPTION
**Pull request by yabinji**

### CODE REVIEW

File: addons/mogdb/dataprotection/gs-basebackup-backup.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  unset LD_LIBRARY_PATH
+
+  apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y expect
+
+  export LD_LIBRARY_PATH="${OLD_LD_LIBRARY_PATH}"
+fi
+
+/usr/bin/expect <<-EOF
+set time 30
+spawn gs_basebackup -Ft -Pv -c fast -Xf -D ${BACKUP_TMPDIR} -h ${DP_DB_HOST} -U ${DP_DB_USER} -z -W
```
Comment (wangyelei) [inline addons/mogdb/dataprotection/gs-basebackup-backup.sh:23]: it is better to use the compression method in datasafed, the backup speed is very slow. you can reference the `postgresql` addon. and better to stream upload to backupRepo.

Comment (yabinji) [reply addons/mogdb/dataprotection/gs-basebackup-backup.sh:23]: mogdb do not support stream backup and not support pipe password

File: addons/mogdb/dataprotection/gs-basebackup-restore.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+if [ $(remote_file_exists "base.tar.gz") == "true" ]; then
+  datasafed pull "base.tar.gz" - | tar -xzvf - -C "${DATA_DIR}/"
+elif [ $(remote_file_exists "base.tar") == "true" ]; then
+  datasafed pull "base.tar" - | tar -xvf - -C "${DATA_DIR}/"
+fi
+if [ $(remote_file_exists "pg_wal.tar.gz") == "true" ]; then
+  datasafed pull "pg_wal.tar.gz" - | tar -xzvf - -C "${DATA_DIR}/pg_wal/"
+elif [ $(remote_file_exists "pg_wal.tar") == "true" ]; then
+  datasafed pull "pg_wal.tar" - | tar -xvf - -C "${DATA_DIR}/pg_wal/"
+fi
```
Comment (wangyelei) [inline addons/mogdb/dataprotection/gs-basebackup-restore.sh:34]: these lines are not necessary

File: addons/mogdb/dataprotection/gs-basebackup-backup.sh (hunk 1 above)
Comment (wangyelei) [reply addons/mogdb/dataprotection/gs-basebackup-backup.sh:23]: oh, if mogdb support PGPASSWORD enviroment, you can use datasafed to stream upload the base.tar by pipe.

Comment (yabinji) [reply addons/mogdb/dataprotection/gs-basebackup-backup.sh:23]: 1. Also not support PGPASSWORD enviroment.
2. gs_basebackup command has a limitation that a valid directory must be provided('-' is not valid).
3. Do you mean to run follow command:
	1) spawn gs_basebackup -Ft -Pv -c fast -Xf -D ${BACKUP_TMPDIR} -h ${DP_DB_HOST} -U ${DP_DB_USER} -W
	2) datasafed push -z zstd-fastest - "/${DP_BACKUP_NAME}.tar.zst" < "$BACKUP_TAR_FILE"

File: addons/mogdb/dataprotection/gs-basebackup-restore.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  exit 0
+fi
+
+
+# NOTE: restore from an old version backup, will be removed in 0.8
+if [ $(remote_file_exists "base.tar.gz") == "true" ]; then
+  datasafed pull "base.tar.gz" - | tar -xzvf - -C "${DATA_DIR}/"
+elif [ $(remote_file_exists "base.tar") == "true" ]; then
+  datasafed pull "base.tar" - | tar -xvf - -C "${DATA_DIR}/"
+fi
```
Comment (wangyelei) [inline addons/mogdb/dataprotection/gs-basebackup-restore.sh:29]: these lines are not necessary too.

**Pull request closed without merging**